src/gfn/env.py

```
@@ -62,7 +62,10 @@ def __init__(
         self.s0 = s0
 
         if sf is None:
-            sf = torch.full(s0.shape, -float("inf"))
+            assert isinstance(s0, torch.Tensor), "When sf is None, s0 must be a Tensor"
+            sf = torch.full(
+                s0.shape, default_fill_value_for_dtype(s0.dtype), dtype=s0.dtype
+            )
```

I am a bit scared about all the cases that don't support -inf, as it may result in silent errors.

For example, an env with integer dtype, we are using the min value, but this may be a valid state (maybe it is even the s0, so we are starting episodes that are already terminated). 

A possible alternative:
We just do `sf = torch.full(s0.shape, -float("inf"), dtype=s0.dtype)`, so we fail loudly when the dtype doesn't support the -inf. We can also try-catch to change the message prompting the user to specify the sf.

---

> We just do sf = torch.full(s0.shape, -float("inf"), dtype=s0.dtype), so we fail loudly when the dtype doesn't support the -inf. We can also try-catch to change the message prompting the user to specify the sf.

I like this idea!

---

A safe alternative would be just making `sf` a required argument for the base `Env` (and maybe for `DiscreteEnv` and `GraphEnv` as well).